docs/src/matrix.md

```diff
@@ -507,31 +507,8 @@ julia> c = M[1, 1]
 
 ### Transpose
 
-```julia
-transpose(::MatElem{T}) where T <: RingElem
-```
-
-Return the transpose of the given matrix.
-
-**Examples**
-
-```jldoctest
-julia> R, t = PolynomialRing(QQ, "t")
-(Univariate Polynomial Ring in t over Rationals, t)
-
-julia> S = MatrixSpace(R, 3, 3)
-Matrix Space of 3 rows and 3 columns over Univariate Polynomial Ring in t over Rationals
-
-julia> A = S([t + 1 t R(1); t^2 t t; R(-2) t + 2 t^2 + t + 1])
-[t + 1       t             1]
-[  t^2       t             t]
-[   -2   t + 2   t^2 + t + 1]
-
-julia> B = transpose(A)
-[t + 1   t^2            -2]
-[    t     t         t + 2]
-[    1     t   t^2 + t + 1]
-
+```@docs
+transpose(x::Mat{T}) where T <: NCRingElement
```

Comment:
I think it complains about `Mat` here. Probably need `Generic.Mat` ? But why is it that anyway, and not MatElem / MatrixElem?

Reply:
I don't think it should be `Mat`. I think you are right.